### PR TITLE
#2079 SMS Content Length StatsD Metric

### PR DESCRIPTION
**Pull request by mchlwellman**
# Description

This PR adds a new metric type to the StatsD client, histogram, which matches the metric that Datadog has added. This metric type is used to track sms content length. In Datadog, normal statsd gauges are flushed every 10 seconds, and only the last value is retained, making it unsuitable for tracking sms content length.

issue #2079 

## How Has This Been Tested?

This was [deployed](https://github.com/department-of-veterans-affairs/notification-api/actions/runs/11962258146) and tested in dev, and perf by deploying and observing the metrics in Datadog.

Screenshot 2024-11-21 at 4 43 42 PM

Screenshot 2024-11-21 at 4 44 11 PM

## Checklist

- [x] I have assigned myself to this PR
- [x] PR has an [appropriate title](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Engineering/team_agreement.md#naming-prs): #9999 - What the thing does
- [x] PR has a detailed description, including links to specific documentation
- [x] I have added the [appropriate labels](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Engineering/pull_request_labels.md#vanotify-labels) to the PR.
- [x] I did not remove any parts of the template, such as checkboxes even if they are not used
- [x] My code follows the [style guidelines](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Process/style_guide.md) of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to any documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works. [Testing guidelines](https://github.com/department-of-veterans-affairs/vanotify-team/blob/main/Process/testing_guide.md)
- [x] I have ensured the latest main is merged into my branch and all checks are green prior to review
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] The ticket was moved into the DEV test column when I began testing this change

### CODE REVIEW

**Review by cris-oddball — CHANGES_REQUESTED** [review]
this cannot have the branch name in the pyproject.toml for utils.

**Review by cris-oddball — DISMISSED** [review]

File: .talismanrc
Comment (EvanParish) [inline .talismanrc:1]: Why are there changes to this file?

**Review by EvanParish — DISMISSED** [review]
It's weird that there are talisman changes, otherwise looks good.

File: .talismanrc
Comment (mchlwellman) [reply .talismanrc:1]: It appears to be sorted. Pre-commit maybe?

**Review by k-macmillan — APPROVED** [review]

File: .talismanrc
Comment (mchlwellman) [reply .talismanrc:1]: When I run `talisman -i -g pre-commit` to add the poetry lock file, it appears to sort these?

**Review by k-macmillan — APPROVED** [review]

File: .talismanrc
Comment (EvanParish) [reply .talismanrc:1]: Yeah, weird. Not really a big deal, but just thought it was odd. It's not anything we need to hold the PR up for.

**Review by EvanParish — APPROVED** [review]

**Pull request merged**